olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -154,75 +153,78 @@ PartitionsResponse PartitionsRepository::GetVolatilePartitions(
     return expiry_response.GetError();
   }
 
-  return GetPartitions(std::move(catalog), std::move(layer),
-                       cancellation_context, std::move(request),
-                       std::move(settings), expiry_response.MoveResult());
+  return GetPartitions(std::move(layer), request, context, boost::none,
+                       expiry_response.MoveResult());
 }
 
 PartitionsResponse PartitionsRepository::GetPartitions(
-    client::HRN catalog, std::string layer,
-    client::CancellationContext cancellation_context, PartitionsRequest request,
-    const client::OlpClientSettings& settings, boost::optional<time_t> expiry) {
-  auto fetch_option = request.GetFetchOption();
-  const auto key = request.CreateKey(layer);
+    std::string layer, const PartitionsRequest& request,
+    client::CancellationContext context, boost::optional<std::int64_t> version,
+    boost::optional<time_t> expiry) {
+  // PartitionsCacheRepository still using deprecated
+  // PartitionsRequest::GetVersion, so remove `request_copy` when the repository
+  // supports version.
+  auto request_copy = request;
+  request_copy.WithVersion(version);
```

Comment:
As mentioned in the comment, `PartitionsCacheRepository` has no version arg yet.
Adding it is a part of another task(Removing `WithVersion()` from `PartitionRequest`), which will be done in the nearest feature.
We are splitting changes, because some files(e.g. PartitionsRepositoryTest.cpp) are changed a lot in both tasks.
The PR makes `PartitionsRepository` support version arg, so we don't have to do similar changes twice.

Reply:
removed